lua/vgit/git/cli/Git.lua

```diff
@@ -8,18 +8,42 @@ local Hunk = require('vgit.git.cli.models.Hunk')
 local Log = require('vgit.git.cli.models.Log')
 local File = require('vgit.git.cli.models.File')
 local Blame = require('vgit.git.cli.models.Blame')
+local git_setting = require('vgit.settings.git')
 
 local Git = Object:extend()
 
+local function table_merge(table1, table2)
+  table1 = vim.deepcopy(table1)
+  for _, value in ipairs(table2) do
+    table1[#table1 + 1] = value
+  end
+  return table1
+end
+
 function Git:constructor(cwd)
-  return {
+  local newself = {
     cwd = cwd or '',
+    cmd = git_setting:get('cmd'),
+    fallback_args = {},
     diff_algorithm = 'myers',
     empty_tree_hash = '4b825dc642cb6eb9a060e54bf8d69288fbee4904',
     state = {
       config = nil,
     },
   }
+  if cwd and not self.is_inside_git_dir(newself) then
+    -- print("cwd", newself.cwd, "->", git_setting:get('fallback_cwd'))
+    newself.cwd = git_setting:get('fallback_cwd') or ''
+    -- cwd must end in spearator for utils.str.strip to work
```

Comment:
I think we should remove these comments, unless this PR is still wip.

Reply:
I considered it WIP until I get more feedback from you if this is the way you wanted it done, with the intention of removing all of this for the final commit.

Reply:
Removed the `print` comment, I'll left the others until you decided what to do with it as I'm still not super happy about the code being this fragile due to `utils.str.strip`.

Reply:
Hmm, yeah `utils.str.strip` is definitely not fully fleshed out. I think we can use `Path.normalize` from plenary as an alternative:

```lua
function fs.normalize_filename(filepath, path)
  return Path:new(filepath):normalize(path)
end
```

And replace `utils.str.strip` with `fs.normalize_filename` in `Git.lua`. I still have to test this out a bit more incase I break some stuff (this reminds me to write more tests for this repo).

Reply:
`plenary.path.normalize` isn't a fit here, it sometimes adds `~` to the the path, I used `make_relative` instead which seems better suited for what `strip` was intended.